From patchwork Thu Sep 20 14:29:26 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andrew Jones <drjones@redhat.com>
X-Patchwork-Id: 10607831
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id F3F6814BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 20 Sep 2018 14:29:30 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E48AB2D976
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 20 Sep 2018 14:29:30 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id D8D8E2D97C; Thu, 20 Sep 2018 14:29:30 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6FE6B2D976
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 20 Sep 2018 14:29:30 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S2387652AbeITUNN (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 20 Sep 2018 16:13:13 -0400
Received: from mx1.redhat.com ([209.132.183.28]:53128 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S2387589AbeITUNN (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 20 Sep 2018 16:13:13 -0400
Received: from smtp.corp.redhat.com (int-mx04.intmail.prod.int.phx2.redhat.com
 [10.5.11.14])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id EE9CF2C9DBD
        for <kvm@vger.kernel.org>; Thu, 20 Sep 2018 14:29:28 +0000 (UTC)
Received: from kamzik.brq.redhat.com (unknown [10.43.2.160])
        by smtp.corp.redhat.com (Postfix) with ESMTP id EE36B874F3;
        Thu, 20 Sep 2018 14:29:27 +0000 (UTC)
From: Andrew Jones <drjones@redhat.com>
To: pbonzini@redhat.com, kvm@vger.kernel.org
Subject: [PATCH kvm-unit-tests] arm/arm64: fix gic-ipi test regression
Date: Thu, 20 Sep 2018 16:29:26 +0200
Message-Id: <20180920142926.13749-1-drjones@redhat.com>
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.14
X-Greylist: Sender IP whitelisted,
 not delayed by milter-greylist-4.5.16 (mx1.redhat.com [10.5.110.29]);
 Thu, 20 Sep 2018 14:29:29 +0000 (UTC)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The gic-ipi tests started crashing in a strange way after
commit 9064d89b ("kvm-unit-tests: unlock before calling
return"). Eventually I narrowed it down to the NR_CPUS change
being the problem. I still don't understand what's going on,
but NR_CPUS=511 works and NR_CPUS=512 doesn't. I don't have
time to debug this further right now, so this is just a
temporary hack that appears to fix the regression.

Signed-off-by: Andrew Jones <drjones@redhat.com>
---
 lib/arm/asm/setup.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/arm/asm/setup.h b/lib/arm/asm/setup.h
index 3215814603e4..81cac019b1d1 100644
--- a/lib/arm/asm/setup.h
+++ b/lib/arm/asm/setup.h
@@ -9,7 +9,7 @@
 #include <asm/page.h>
 #include <asm/pgtable-hwdef.h>
 
-#define NR_CPUS			512
+#define NR_CPUS			511
 extern u64 cpus[NR_CPUS];	/* per-cpu IDs (MPIDRs) */
 extern int nr_cpus;
 
